### PR TITLE
Adding flag to enable restarting running stages (default disabled)

### PR DESCRIPTION
**Pull request by robzienert**
@anotherchrisberry @ajordens @robfletcher PTAL

### CODE REVIEW

File: app/scripts/modules/core/application/config/applicationAttributes.directive.html [hunk 1]
```diff
@@ -26,6 +26,8 @@
             </dd>
             <dt>Instance Port</dt>
             <dd>{{vm.application.attributes.instancePort}}</dd>
+            <dt ng-if="vm.application.attributes.enableRestartRunningExecutions">Pipeline Behavior</dt>
```
Comment (ajordens) [inline app/scripts/modules/core/application/config/applicationAttributes.directive.html]: Fancy ... I wasn't even expecting there to be a UI for this but nice work!

File: app/scripts/modules/core/application/modal/editApplication.html [hunk 2]
```diff
@@ -164,6 +164,16 @@ <h3 data-purpose="modal-header">Edit Application</h3>
         </div>
       </div>
 
+      <div class="form-group row">
+        <div class="col-sm-3 sm-label-right">Pipeline Behavior</div>
+        <div class="col-sm-9">
+          <input type="checkbox"
+                 ng-model="editApp.applicationAttributes.enableRestartRunningExecutions">
+          Enable restarting running pipeline executions
```
Comment (ajordens) [inline app/scripts/modules/core/application/modal/editApplication.html]: Should we use same wording as above ... `Allow restarting running executions`? Or should s/executions/pipelines?

**Comment by ajordens** [comment]
Other than possible wording this LGTM*

* what do I know about javascript?

File: app/scripts/modules/core/application/modal/editApplication.html (hunk 2 above)
Comment (anotherchrisberry) [inline app/scripts/modules/core/application/modal/editApplication.html]: +1 - `Allow restarting running pipelines` is probably better wording in both places.

**Comment by anotherchrisberry** [comment]
Aside from the wording, this LGTM

**Comment by duftler** [comment]
@robzienert Can you please provide some more details about what behavior this controls? The PR description refers to restarting running stages, but the changes refer to restarting running pipelines.
And once enable, how would I restart a pipeline? Will there be a new control in the ui, or does this affect some backend behavior?

**Comment by ajordens** [comment]
@duftler Right now you can restart a failed stage (_ie. Jenkins_) while the pipeline is still running (especially if the Jenkins stage was `FAILED_CONTINUE`) ... this is generally not recommended behavior and should be gated in `deck`.

We have one internal user with large pipelines that _needs_ to restart a running pipeline hence this flag.

**Pull request merged**